ext/mysql2/statement.c

```diff
@@ -266,8 +291,22 @@ static VALUE execute(int argc, VALUE *argv, VALUE self) {
           break;
         case T_BIGNUM:
           bind_buffers[i].buffer_type = MYSQL_TYPE_LONGLONG;
-          bind_buffers[i].buffer = xmalloc(sizeof(long long int));
-          *(LONG_LONG*)(bind_buffers[i].buffer) = rb_big2ll(argv[i]);
+          {
+              LONG_LONG num;
+              if (my_big2ll(argv[i], &num)) {
```

Comment:
It wasn't immediately clear to me why `my_big2ll` would return "truthy" and then proceed on the workaround path. Maybe flip the cases around? I think this would be more readable:

``` C
if (my_big2ll(argv[i], &num) == 0) {
                bind_buffers[i].buffer_type = MYSQL_TYPE_LONGLONG;
                bind_buffers[i].buffer = xmalloc(sizeof(long long int));
                *(LONG_LONG*)(bind_buffers[i].buffer) = num;
} else {
    /* The bignum was larger than we can fit in LONG_LONG, send it as a string */
    ... code as written ...
}
```